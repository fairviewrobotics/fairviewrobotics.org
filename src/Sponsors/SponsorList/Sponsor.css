.company {
  margin: 3vw;
  opacity: 0.8;
  transition: opacity 0.25s;
}

.company:hover {
  opacity: 1;
}

@media(max-width:768px) {
  .company {
    margin: 15vw;
    transform: scale(1.8);
  }
}