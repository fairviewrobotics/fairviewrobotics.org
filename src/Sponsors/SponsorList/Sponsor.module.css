.sponsor {
  margin: 3vw;
  opacity: 0.8;
  transition: opacity 0.25s;
}

.sponsor:hover {
  opacity: 1;
}

.image {
  height: auto;
}

@media(max-width:768px) {
  .sponsor {
    margin: 15vw;
    transform: scale(1.8);
  }
}