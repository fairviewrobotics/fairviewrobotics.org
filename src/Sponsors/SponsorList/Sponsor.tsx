import React, { PureComponent } from 'react';

import styles from './Sponsor.module.css';

export type SponsorProps = {
  name: string;
  size: number;
  url: string;
  src: string;
}

export default class Sponsor extends PureComponent<SponsorProps> {
  render() {
    const trueWidth = this.props.size * .9;

    return (
      <a className={styles.sponsor} target='_blank' href={this.props.url} rel="noopener noreferrer">
        <img src={this.props.src} alt={this.props.name} className={styles.image} style={{ width: `${trueWidth}vw` }} />
      </a>
    );
  }
}
