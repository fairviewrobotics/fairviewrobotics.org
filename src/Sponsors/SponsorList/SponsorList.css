#companies-outer {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  /* TODO: make BackgroundImage inline so margin-top is not needed */
  margin-top: 25vw;
}