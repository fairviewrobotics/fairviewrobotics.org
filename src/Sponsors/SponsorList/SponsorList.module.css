.sponsorList {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
  width: 75%;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
}