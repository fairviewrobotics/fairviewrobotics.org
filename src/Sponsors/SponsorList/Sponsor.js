import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './Sponsor.css';

export default class Sponsor extends PureComponent {

  static propTypes = {
    name: PropTypes.string.isRequired,
    size: PropTypes.number.isRequired,
    url: PropTypes.string.isRequired,
    src: PropTypes.string.isRequired
  };

  render() {
    const trueWidth = this.props.size * .9;

    return (
      <a className='company' target='_blank' href={this.props.url}>
        <img src={this.props.src} alt={this.props.name} style={{ height: 'auto', width: `${trueWidth}vw` }} />
      </a>
    );
  }
}
