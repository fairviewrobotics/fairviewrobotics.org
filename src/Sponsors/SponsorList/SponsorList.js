import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import Sponsor from "./Sponsor";

import './SponsorList.css';

export default class SponsorList extends PureComponent {

  static propTypes = {
    sponsors: PropTypes.arrayOf(PropTypes.shape({
      name: PropTypes.string.isRequired,
      size: PropTypes.number.isRequired,
      url: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired
    })).isRequired
  };

  render() {
    return (
      <div id="companies-outer">
        {this.props.sponsors.map(sponsor =>
          <Sponsor key={sponsor.name} name={sponsor.name} size={sponsor.size} url={sponsor.url} src={sponsor.src} />
        )}
      </div>
    );
  }
}
