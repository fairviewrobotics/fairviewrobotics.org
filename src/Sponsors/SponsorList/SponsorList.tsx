import React, { PureComponent } from 'react';
import Sponsor from "./Sponsor";

import styles from './SponsorList.module.css';

export type SponsorListProps = {
  sponsors: {
    name: string;
    size: number;
    url: string;
    src: string;
  }[];
}

export default class SponsorList extends PureComponent<SponsorListProps> {
  render() {
    return (
      <div className={styles.sponsorList}>
        {this.props.sponsors.map(sponsor =>
          <Sponsor key={sponsor.name} name={sponsor.name} size={sponsor.size} url={sponsor.url} src={sponsor.src} />
        )}
      </div>
    );
  }
}
