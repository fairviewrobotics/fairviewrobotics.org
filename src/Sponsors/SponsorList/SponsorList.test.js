import React from "react";
import * as ReactDOM from "react-dom";
import { shallow } from "enzyme";

import SponsorList from "./SponsorList";

describe('SponsorList', () => {
  let wrapper;

  it("renders without crashing", () => {
    const sponsors = [];

    const div = document.createElement("div");
    ReactDOM.render(
      <SponsorList sponsors={sponsors}/>,
      div
    );
  });

  describe('when given no `sponsors`', () => {
    beforeEach(() => {
      const sponsors = [];

      wrapper = shallow(
        <SponsorList sponsors={sponsors} />
      );
    });

    it('should render an empty list', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('when given some `sponsors`', () => {
    beforeEach(() => {
      const sponsors = [
        {
          name: 'Cheez it',
          size: 90,
          url: 'http://www.urltocheese.com/',
          src: 'http://www.cheeseimage.com/'
        },
        {
          name: 'Mcdonalds',
          size: 30,
          url: 'http://www.urltomd.com/',
          src: 'http://www.mdimage.com/'
        },
        {
          name: 'Chef Boyardee',
          size: 50,
          url: 'http://www.urltoboyardee.com/',
          src: 'http://www.boyardeeimage.com/'
        }
      ];

      wrapper = shallow(
        <SponsorList sponsors={sponsors} />
      );
    });

    it('should render each sponsor', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

});