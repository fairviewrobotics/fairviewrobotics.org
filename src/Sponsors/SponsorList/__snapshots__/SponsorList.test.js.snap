// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SponsorList when given no \`sponsors\` should render an empty list 1`] = `
<div
  className="sponsorList"
/>
`;

exports[`SponsorList when given some \`sponsors\` should render each sponsor 1`] = `
<div
  className="sponsorList"
>
  <Sponsor
    key="Cheez it"
    name="Cheez it"
    size={90}
    src="http://www.cheeseimage.com/"
    url="http://www.urltocheese.com/"
  />
  <Sponsor
    key="Mcdonalds"
    name="Mcdonalds"
    size={30}
    src="http://www.mdimage.com/"
    url="http://www.urltomd.com/"
  />
  <Sponsor
    key="Chef Boyardee"
    name="Chef Boyardee"
    size={50}
    src="http://www.boyardeeimage.com/"
    url="http://www.urltoboyardee.com/"
  />
</div>
`;
